packages/material-ui/src/Step/Step.test.js

```diff
@@ -1,107 +1,151 @@
   });
 
-  after(() => {
-    mount.cleanUp();
-  });
-
   describeConformance(<Step />, () => ({
     classes,
     inheritComponent: 'div',
     mount,
     refInstanceof: window.HTMLDivElement,
     skip: ['componentProp'],
+    after: () => mount.cleanUp(),
   }));
 
   it('merges styles and other props into the root node', () => {
-    const wrapper = shallow(
+    const { getByTestId } = render(
       <Step
         index={1}
         style={{ paddingRight: 200, color: 'purple', border: '1px solid tomato' }}
-        data-role="Menuitem"
+        data-testid="root"
         orientation="horizontal"
       />,
     );
-    const props = wrapper.props();
-    assert.strictEqual(props.style.paddingRight, 200);
-    assert.strictEqual(props.style.color, 'purple');
-    assert.strictEqual(props.style.border, '1px solid tomato');
-    assert.strictEqual(props['data-role'], 'Menuitem');
+
+    const rootNode = getByTestId('root');
+    expect(rootNode.style).to.have.property('paddingRight', '200px');
+    expect(rootNode.style).to.have.property('color', 'purple');
+    expect(rootNode.style).to.have.property('border', '1px solid tomato');
   });
 
   describe('rendering children', () => {
     it('renders children', () => {
-      const children = <h1 className="hello-world">Hello World</h1>;
-      const wrapper = shallow(
-        <Step label="Step One" index={1} orientation="horizontal">
-          {children}
+      const { getByTestId } = render(
+        <Step data-testid="root" label="Step One" index={1} orientation="horizontal">
+          <StepChildDiv data-testid="child">Hello World</StepChildDiv>
         </Step>,
       );
-      assert.strictEqual(wrapper.find('.hello-world').length, 1);
+
+      expect(within(getByTestId('root')).getByTestId('child')).to.be.ok;
     });
 
     it('renders children with all props passed through', () => {
-      const children = [
-        <h1 key={1} className="hello-world">
-          Hello World
-        </h1>,
-        <p key={2} className="hay">
-          How are you?
-        </p>,
-      ];
-      const wrapper = shallow(
+      const { getAllByTestId } = render(
         <Step active={false} completed disabled index={0} orientation="horizontal">
-          {children}
+          <PropsAsDataset />
+          <PropsAsDataset />
         </Step>,
       );
-      const child1 = wrapper.find('.hello-world');
-      const child2 = wrapper.find('.hay');
-      [child1, child2].forEach(child => {
-        assert.strictEqual(child.length, 1);
-        assert.strictEqual(child.props().active, false);
-        assert.strictEqual(child.props().completed, true);
-        assert.strictEqual(child.props().disabled, true);
-        assert.strictEqual(child.props().icon, 1);
+      getAllByTestId('props').forEach(child => {
+        // HTMLElement.dataset is a DOMStringMap which fails deep.equal
+        const datasetAsObject = { ...child.dataset };
+        expect(datasetAsObject).to.deep.equal({
```

Comment:
Using deep equality over (previous) subset checking. This helps exposing all props that children might receive.